client/src/App.js

```diff
@@ -1,12 +1,61 @@
 import "./App.css";
-
+import { useState, useEffect } from "react";
+import AddAndSearch from "./Components/AddAndSearch";
+import Footer from "./Components/Footer";
+import NavBar from "./Components/NavBar";
+import VideoList from "./Components/VideoList";
+import Welcome from "./Components/Welcome";
+import AddVideo from "./Components/AddVideo";
+// import exampleresponse from "./exampleresponse.json";
 function App() {
+  const [keyword, setKeyword] = useState("");
+  const [videoCards, setVideoCards] = useState([]);
+  const [error, setError] = useState(null);
+  const [loading, setLoading] = useState(true);
+  useEffect(() => {
+    fetchVideos();
+  }, []);
+
+  const fetchVideos = async () => {
+    try {
+      const response = await fetch(
+        "https://video-recomendations-server.onrender.com/"
+      );
+      if (!response.ok) {
+        throw Error(
+          `The fetching of videos was not successful. Error: ${response.status}`
+        );
+      }
+      const data = await response.json();
+      setVideoCards(data);
+      setLoading(false);
+      setError(null);
+    } catch (error) {
+      setError(error.message);
+      setLoading(false);
+    }
+  };
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <h1>Video Recommendation</h1>
-      </header>
-    </div>
+    <>
+      <NavBar />
+      <Welcome />
+      <AddAndSearch setKeyword={setKeyword} />
+      <AddVideo setVideoCards={setVideoCards} videoCards={videoCards} />
```

Comment:
The logic works, one suggestion that helps improve the readability, we can write the if, else if and else in this way so that it is easier to read

```
if(isLoading){
   return (
     <h1 className="blink_me">Loading.....</h1>
)

if(error) {
  return ( <h1 className="error-msg "> {error}</h1>)
}
  // main logic
return (
     <VideoList ... />
)
}
```

Reply:
Thanks, I'll make this change.